hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/MergeIntoHoodieTableCommand.scala

```diff
@@ -515,7 +521,16 @@ case class MergeIntoHoodieTableCommand(mergeInto: MergeIntoTable) extends Hoodie
         HoodieWriteConfig.INSERT_PARALLELISM_VALUE.key -> hoodieProps.getString(HoodieWriteConfig.INSERT_PARALLELISM_VALUE.key, "200"), // set the default parallelism to 200 for sql
         HoodieWriteConfig.UPSERT_PARALLELISM_VALUE.key -> hoodieProps.getString(HoodieWriteConfig.UPSERT_PARALLELISM_VALUE.key, "200"),
         HoodieWriteConfig.DELETE_PARALLELISM_VALUE.key -> hoodieProps.getString(HoodieWriteConfig.DELETE_PARALLELISM_VALUE.key, "200"),
-        SqlKeyGenerator.PARTITION_SCHEMA -> partitionSchema.toDDL
+        SqlKeyGenerator.PARTITION_SCHEMA -> partitionSchema.toDDL,
+
+        // NOTE: We have to explicitly override following configs to make sure no schema validation is performed
+        //       as schema of the incoming dataset might be diverging from the table's schema (full schemas'
+        //       compatibility b/w table's schema and incoming one is not necessary in this case since we can
+        //       be cherry-picking only selected columns from the incoming dataset to be inserted/updated in the
+        //       target table, ie partially updating)
+        AVRO_SCHEMA_VALIDATE_ENABLE.key -> "false",
+        RECONCILE_SCHEMA.key -> "false",
```

Comment:
does this mean that, w/ MERGE INTO, we don't support reconciling schema ? 
I am w/ you on the intent. trying to gauge what all are not supported w/ MERGE INTO wrt schema evolution. do we expect users to always go through ALTER TABLE then?

Reply:
Yeah, we can't support reconciliation due to the nature of the feature (incoming batch is essentially an inner-join of existing table and the incoming records)